### DISCUSSION TITLE
Using `Bytes` on a `&Vec<u8>` field

### DISCUSSION THREAD

**Discussion by theastrallyforged** [jonasbb/serde_with#460] [Q&A] [answered]
.hg If you have, say, macro-generated code that serializes borrowed values, you can't currently use `serde_with::Bytes` with it (or you have to specialcase `Vec<u8>` in the macro and map it to `[u8]`)

I did take a shot at implementing this myself, since I figured it'd be pretty simple, but I got stuck at adding a test for it: the test assumes that `DeserializeAs` is implemented as well as `SerializeAs`, and while you can deserialize a `&[u8]`, I'm don't think you can deserialize a `&Vec<u8>`

**Comment by jonasbb** [comment, marked as answer]
In general, the type you write inside the `serde_as` attribute should mirror the field type. The only exceptions are the places where you want to change the behavior, such as for `Vec<u8> -> Bytes` or `BTreeMap<X, Y> -> Vec<(X, Y)>`. This also applies to references `&` which should be mirrored too.

This means you can write:

```rust
#[serde_as(as = "&Bytes")]
vec: &'a Vec<u8>,
```

This is possible due to the generic `SerializeAs` implementation for `&T`:
https://github.com/jonasbb/serde_with/blob/9e952e2375a9963a187c12ffaf03e3d794539d6e/serde_with/src/ser/impls.rs#L32
